core/src/main/java/org/elasticsearch/indices/IndexingMemoryController.java

```diff
@@ -201,17 +201,23 @@ protected void deactivateThrottling(IndexShard shard) {
 
     @Override
     public void postIndex(Engine.Index index, Engine.IndexResult result) {
-        recordOperationBytes(result);
+        recordOperationBytes(index, result);
     }
 
     @Override
     public void postDelete(Engine.Delete delete, Engine.DeleteResult result) {
-        recordOperationBytes(result);
+        recordOperationBytes(delete, result);
     }
 
     /** called by IndexShard to record that this many bytes were written to translog */
-    private void recordOperationBytes(Engine.Result result) {
-        statusChecker.bytesWritten(result.getSizeInBytes());
+    private void recordOperationBytes(Engine.Operation operation, Engine.Result result) {
+        final int sizeInBytes;
+        if (result.getTranslogLocation() != null) {
```

Comment:
I talked this one with @mikemccand and since this whole thing is a heuristic to approximate the memory usage of the IW, we think it's OK to always use the `operation.estimatedSizeInBytes()` . No need to look at the translog location.